### DISCUSSION TITLE
Automation: Retrieve data from Query Rows step

### DISCUSSION THREAD

**Discussion by xnohat** [Budibase/budibase#5048] [Help] [answered]
Hi
I found an bug in version 1.0.90 Automation

1. I create Automation with 1st step is Query Rows, get all rows of table
![image](https://user-images.githubusercontent.com/4613808/159764584-81aad270-2762-4866-8532-35c0d196bd71.png)
2. I using JS Scripting block for manipulate data from Query Rows step
```
var rows = {{steps.1.rows}}; 
return rows;
```

cannot access steps.1.rows without {{ }}

3. Test Automation. Problem appear: Data from Query Rows only return [object Object],[object Object],[object Object] 
![image](https://user-images.githubusercontent.com/4613808/159764847-8ec218e1-13d5-43cc-80ff-36192ca14919.png)

I have tried again, in step Query Rows, set Limit to 1 , Query Rows will return data normal in JSON form

Anyone can help me confirm this is a bug ?

**Comment by mjashanks** [comment, marked as answer]
Hey @xnohat  - i agree this is confusing. I have logged this already in here: https://github.com/Budibase/budibase/issues/4661

You can see the workaround in there - you should use `steps[1].rows`

**Reply by xnohat** [reply]
Many thanks, you saved my works !
